### PR TITLE
gcp: vm module

### PR DESCRIPTION
**Pull request by ggramal**
Interface looks like this
```
locals {
  vms = {
    proxy = {
      name         = "proxy"
      machine_type = "e2-micro"
      network_config = {
        subnet = module.gcp.subnets["main"]["vms"].name
        public_ip = {
          ephemeral = true
        }
      }
      image = {
        project = "cos-cloud"
        family  = "cos-stable"
      }
      zone = var.gcp_projects.prd.default_zone
      metadata = {
        user-data = <<EOF
#cloud-config
runcmd:
  - docker run --rm -d -p 45080:8888 ghcr.io/jee-r/tinyproxy:76151017
EOF
      }
      service_account = {
        name   = "proxysa"
        roles  = []
        scopes = []
      }
    }
  }
}
```

### CODE REVIEW

File: modules/gcp/main.tf [hunk 1]
```diff
@@ -361,3 +361,29 @@ module "firestores" {
...
+  source       = "./vm"
+  for_each     = var.vms
+  project_id   = var.project.id
+  name         = each.value.name
+  machine_type = each.value.machine_type
+  disk = try(
+    each.value.disk,
+    {
+      size_gb = 10,
+      type    = "pd-standard"
```
Comment (devops-dmitry) [inline modules/gcp/main.tf:377]: can we use pd-balanced by default?

File: modules/gcp/main.tf [hunk 2]
```diff
@@ -361,3 +361,29 @@ module "firestores" {
...
   datastore_indices = try(each.value.datastore_indices, {})
 }
+
+/* vms */
+
+module "vms" {
+  source       = "./vm"
+  for_each     = var.vms
+  project_id   = var.project.id
+  name         = each.value.name
```
Comment (devops-dmitry) [inline modules/gcp/main.tf:371]: maybe it's better not to duplicate the name, let's use name=each.key

File: modules/gcp/main.tf (hunk 1 above)
Comment (ggramal) [reply modules/gcp/main.tf:377]: yes. let me change

File: modules/gcp/main.tf (hunk 2 above)
Comment (devops-dmitry) [reply modules/gcp/main.tf:371]: ok. it 's not important

**Review by devops-dmitry — APPROVED** [review]

**Pull request merged**